pkg/builder/builder.go

```diff
@@ -862,20 +862,30 @@ func makeScrutinizeInitContainers(vscr *v1beta1.VerticaScrutinize, vdb *vapi.Ver
 
 // BuildVProxyDeployment builds manifest for a subclusters VProxy deployment
 func BuildVProxyDeployment(nm types.NamespacedName, vdb *vapi.VerticaDB, sc *vapi.Subcluster) *appsv1.Deployment {
+	proxyLabels := MakeLabelsForVProxyObject(vdb, sc, true)
+	depSelectorLabels := MakeDepSelectorLabels(vdb, sc)
+	// pod should have both proxy labels and deployment selector labels
+	depPodLabels := make(map[string]string)
+	for k, v := range proxyLabels {
+		depPodLabels[k] = v
+	}
+	for k, v := range depSelectorLabels {
+		depPodLabels[k] = v
+	}
```

Comment:
If you do what I mentioned above, you won't need this, you can set vmeta.ProxyPodSelectorLabel in MakeCommonLabels

Reply:
I don't see how MakeCommonLabels will simplify the code here. The proxy labels are for proxy deployment, here I am collecting labels for proxy pods.

Reply:
I made the changes in makeLabelsForObject which is the only caller of MakeCommonLabels.